src/cuga/backend/knowledge/auth.py

```
@@ -0,0 +1,214 @@
+                roles=None,
+            )
+
+    # --- External mode (browser/SDK) ---
+    # Check if auth is enabled
+    from cuga.backend.server.auth.dependencies import _auth_enabled, get_current_user
+
+    current_user = None
+    if _auth_enabled():
+        try:
+            current_user = await get_current_user(request)
+        except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)